windows/ReactNativeCameraCPP/ReactCameraView.cpp

```diff
@@ -0,0 +1,537 @@
+    {
+        m_unloadedEventToken.revoke();
+    }
+
+    void ReactCameraView::Initialize()
+    {
+        m_childElement = winrt::CaptureElement();
+        Children().Append(m_childElement);
+        // RNW does not support DropView yet, so we need to manually register to Unloaded event and remove self
+        // from the static view list
+        m_unloadedEventToken = Unloaded(winrt::auto_revoke, [ref = get_weak()](auto const&, auto const&) {
+            if (auto self = ref.get()) {
+                auto unloadedAction{ self->OnUnloaded() };
+                unloadedAction.Completed([self](auto&& /*sender*/, AsyncStatus const /* args */) {
+                    winrt::ReactNativeCameraCPP::implementation::ReactCameraViewManager::RemoveViewFromList(self);
+                });
+            }
+        });
+    }
+
+    void ReactCameraView::UpdateProperties(IJSValueReader const& propertyMapReader)
+    {
+        const JSValueObject& propertyMap = JSValue::ReadObjectFrom(propertyMapReader);
+
+        for (auto const& pair : propertyMap) {
+            auto const& propertyName = pair.first;
+            auto const& propertyValue = pair.second;
+            if (!propertyValue.IsNull()) {
+                if (propertyName == "torchMode") {
+                    UpdateTorchMode(static_cast<int>(propertyValue.Double()));
+                }
+                if (propertyName == "flashMode") {
+                    UpdateFlashMode(static_cast<int>(propertyValue.Double()));
+                }
+                else if (propertyName == "type") {
+                    UpdateDeviceType(static_cast<int>(propertyValue.Double()));
+                }
+                else if (propertyName == "keepAwake") {
+                    UpdateKeepAwake(propertyValue.Boolean());
+                }
+                else if (propertyName == "aspect") {
+                    UpdateAspect(static_cast<int>(propertyValue.Double()));
+                }
+            }
+        }
+    }
+
+    IAsyncAction ReactCameraView::UpdateFilePropertiesAsync(StorageFile storageFile, std::map<std::wstring, JSValue> const& options)
+    {
+        auto props = co_await storageFile.Properties().GetImagePropertiesAsync();
+        auto searchTitle = options.find(L"title");
+        if (searchTitle != options.end()) {
+            const auto& titleValue = options.at(L"title");
+            if (titleValue.Type() == JSValueType::String)
+            {
+                auto titleString = titleValue.String();
+                props.Title(winrt::to_hstring(titleString));
+            }
+            co_await props.SavePropertiesAsync();
+        }
+    }
+    
+    // RNW has a bug where the numeric value is set as int in debug but double in release
+    // ToDo: remove this function once bug https://github.com/microsoft/react-native-windows/issues/4225 is fixed.
+    bool ReactCameraView::GetNumericValue(std::map<std::wstring, JSValue> const& options, const std::wstring key, int &value)

```

Comment:
>GetNumericValue [](start = 26, length = 15)

TryGetInt ? or TryGetValueAsInt

Reply:
Sounds good!

Reply:
looks like a typo snuck in by 😛

---
In reply to: [389814407](https://github.com/react-native-community/react-native-camera/pull/2723#discussion_r389814407) [](ancestors = 389814407)